torchvision/datasets/_optical_flow.py

```diff
@@ -363,6 +364,73 @@ def _read_flow(self, file_name):
         return _read_pfm(file_name)
 
 
+class HD1K(FlowDataset):
+    """`HD1K <http://hci-benchmark.iwr.uni-heidelberg.de/>`__ dataset for optical flow.
+
+    The dataset is expected to have the following structure: ::
+
+        root
+            hd1k
+                hd1k_challenge
+                    image_2
+                hd1k_flow_gt
+                    flow_occ
+                hd1k_input
+                    image_2
+
+    Args:
+        root (string): Root directory of the HD1K Dataset.
+        split (string, optional): The dataset split, either "train" (default) or "test"
+        transforms (callable, optional): A function/transform that takes in
+            ``img1, img2, flow, valid`` and returns a transformed version.
+    """
+
+    _has_builtin_flow_mask = True
+
+    def __init__(self, root, split="train", transforms=None):
+        super().__init__(root=root, transforms=transforms)
+
+        verify_str_arg(split, "split", valid_values=("train", "test"))
+
+        root = Path(root) / "hd1k"
+        if split == "train":
+            # There are 36 "sequences" and we don't want seq i to overlap with seq i + 1, so we need this for loop
+            for seq_idx in range(36):
+                flows = sorted(glob(str(root / "hd1k_flow_gt" / "flow_occ" / f"{seq_idx:06d}_*.png")))
+                images = sorted(glob(str(root / "hd1k_input" / "image_2" / f"{seq_idx:06d}_*.png")))
+                for i in range(len(flows) - 1):
+                    self._flow_list += [flows[i]]
+                    self._image_list += [[images[i], images[i + 1]]]
+        else:
+            images1 = sorted(glob(str(root / "hd1k_challenge" / "image_2" / "*10.png")))
+            images2 = sorted(glob(str(root / "hd1k_challenge" / "image_2" / "*11.png")))
```

Comment:
That seems like an odd globbing pattern. Do all images actually end on `10.png` and `11.png`?

Reply:
Yes, this directory contains the following images:

```
100000_0010.png  100003_0010.png  100006_0010.png  100009_0010.png  100012_0010.png  100015_0010.png  100018_0010.png  100021_0010.png  100024_0010.png
100000_0011.png  100003_0011.png  100006_0011.png  100009_0011.png  100012_0011.png  100015_0011.png  100018_0011.png  100021_0011.png  100024_0011.png
100001_0010.png  100004_0010.png  100007_0010.png  100010_0010.png  100013_0010.png  100016_0010.png  100019_0010.png  100022_0010.png  100025_0010.png
100001_0011.png  100004_0011.png  100007_0011.png  100010_0011.png  100013_0011.png  100016_0011.png  100019_0011.png  100022_0011.png  100025_0011.png
100002_0010.png  100005_0010.png  100008_0010.png  100011_0010.png  100014_0010.png  100017_0010.png  100020_0010.png  100023_0010.png  100026_0010.png
100002_0011.png  100005_0011.png  100008_0011.png  100011_0011.png  100014_0011.png  100017_0011.png  100020_0011.png  100023_0011.png  100026_0011.png
```

The first part is a "scene" id and the second part identifies the first and second image of that scene. There are only 2 images per scene so this globbing pattern does the job.

We already have a similar pattern in KittiFlow BTW

https://github.com/pytorch/vision/blob/22ff44fd14139f2a056ad52b9bd109bd958089f3/torchvision/datasets/_optical_flow.py#L180-L181